### DISCUSSION TITLE
error calling Action in store with nuxt-socket.io err_action_access_undefined - Nuxt 2

### DISCUSSION THREAD

**Discussion by luansantosmx** [nuxt/nuxt#10332] [Questions (Nuxt 2)]
Hello guys!

I'm having problems calling actions in the store when I configure the nuxt-socket.io in the project, I'm configuring the nuxt-socket.io in the nuxt.config.js and calling the connection in the component according to the nuxt-socket.io documentation ( [https://nuxt-socket-io.netlify.app/usage](url) ), but when this configuration is performed, the components can no longer call the actions of the store.

I'm using vuex and vuex-module-decorators.

Config nuxt.config.js:

    io: {
        sockets: [
            {
                name: 'home',
                url: 'http://url-server-backend.com',
                default: true,
            }
        ]
    },

Config component:

    async beforeMount(){
        const token = this.$cookies.get('authToken');
        this.socket = this.$nuxtSocket({
            name: 'home',
            channel: '/socket-io',
            persist: true,
            teardown: true,
            reconnection: false,
            extraHeaders: {
                Authorization: token
            }
        });
    },

Error:

![photo_2022-03-15 09 23 34](https://user-images.githubusercontent.com/78658979/158385235-313fb98d-3703-4aea-8ad4-b2aba5011b7c.jpeg)

can you please help me?

**Comment by Jeanlucascastro** [comment]
having this problem too

**Comment by burakovevgeny** [comment]
I solved this problem when I moved the connection ws logic to the created hook
